File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.py
def solution(n, left, right):
    answer = []
    R = left // n
    C = left % n

    while left <= right:

        answer.append(max(R, C) + 1)

        C += 1

        if C == n:
            C = 0
            R += 1

        left += 1

    return answer

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(int n, long long left, long long right) {
    vector<int> answer;
    long long R = left / n;
    long long C = left % n;

    while (left <= right) {
        answer.push_back(max(R, C) + 1);

        C++;
        if (C == n) {
            C = 0;
            R++;
        }

        left++;
    }

    return answer;
}
